controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +841,142 @@ var _ = Describe("reconcileRateLimit", func() {
 		Expect(reconcileRateLimit(hetznerCluster, testEnv.RateLimitWaitTime)).To(BeFalse())
 	})
 })
+
+func TestSetControlPlaneEndpoint(t *testing.T) {
+	t.Run("return false and don't make changes to ControlPlaneEndpoint if load balancer is not enabled and ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{
+					Enabled: false,
+				}, 
+				ControlPlaneEndpoint: nil,
+			},
+		}
+		got := setControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
+			t.Fatalf("ControlPlaneEndpoint must be nil")
+		}
+		if got != false {
+			t.Fatal("return value should be false")
+		}
+	})
+
+	t.Run("return true and don't make changes to ControlPlaneEndpoint if load balancer is not enabled but ControlPlaneEndpoint is not nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{
+					Enabled: false,
+				}, 
+				ControlPlaneEndpoint: &clusterv1.APIEndpoint{
+					Host: "xyz",
+					Port: 12,
+				},
+			},
+		}
+		got := setControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
+			t.Fatalf("ControlPlaneEndpoint must not be nil")
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Host != "xyz"){
+			t.Fatalf("Wrong input for host. Got: %s, Want: 'xyz'", hetznerCluster.Spec.ControlPlaneEndpoint.Host)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Port != 12){
+			t.Fatalf("Wrong input for port. Got: %d, Want: 12", hetznerCluster.Spec.ControlPlaneEndpoint.Port)
+		}
+		if got != true {
+			t.Fatalf("return value should be true")
+		}
+	})
+
+	t.Run("return false if load balancer is enabled but IPv4 is '<nil>'. ControlPlaneEndpoint should not change", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{
+					Enabled: true,
+				}, 
+				ControlPlaneEndpoint: nil,
+			}, 
+			Status: infrav1.HetznerClusterStatus{
+				ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{
+					IPv4: "<nil>",
+				},
+			},
+		}
+		got := setControlPlaneEndpoint(hetznerCluster)
+		// ControlPlaneEnpdoint should still remain nil
+		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
+			t.Fatalf("ControlPlaneEndpoint should not change. It should remain nil")
+		}
+		if got != false {
+			t.Fatalf("return value should be false")
+		}
+	})
+
+	t.Run("return true if load balancer is enabled and IPv4 is not nil, but ControlPlaneEndpoint is nil. Values of ControlPlaneEndpoint.Host and ControlPlaneEndpoint.Port will get updated", func(t *testing.T) {
```

Comment:
better message: 
... if load balancer is enabled, IPv4 is not nil, and ControlPlaneEndpoint is nil